### ISSUE TITLE
donot restart pod when upgrade the operator

### ISSUE THREAD

**Issue by zlcnju** [zalando/postgres-operator#2214]
when upgrade operator, the PG statefulset will be restarted. It is better to give the recreate access to the user. And I find the [percona give a design with crversion to solve the problem](https://github.com/percona/percona-xtradb-cluster-operator/blob/0fbb9700a5467512871d9c255677f120b7ea790c/pkg/controller/pxc/controller.go#L600). In their solution, the process is:
1. upgrade the operator and the statesulset will not restarted.
2. user change the crversion when they want the new feature
3. statefulset is restarted and new feature is used

**Comment by FxKu** [comment]
I disagree. Isn't automation what users expect from the operator. You update the specification and the operator reflects it. I think it would create more confusion if somebody can not only check the manifest for the current state but must inspect the revision number, too. Sounds tedious. If you don't need the feature now, don't change the manifest.
